flake.nix

```diff
@@ -310,6 +317,13 @@
             echo "doc manual $doc/share/doc/nix/manual" >> $doc/nix-support/hydra-build-products
           '';
 
+          preInstallCheck = ''
+            export PATH=${runCommand "nix-2.3-prefetch" { } ''
+              mkdir -p $out/bin
+              cp ${nixStable}/bin/nix-prefetch-url $out/bin/nix-prefetch-url-2.3
+              $out/bin/nix-prefetch-url-2.3 --version | grep -q 2.3
+            ''}/bin:$PATH
+          '';
```

Comment:
This isn't very pretty, but I wanted to hack together a regression-test now. Any suggestions on how to implement that a little bit nicer? :)